crates/services/txpool/src/containers/dependency.rs

```diff
@@ -268,7 +222,34 @@ impl Dependency {
         let mut max_depth = 0;
         let mut db_coins: HashMap<UtxoId, CoinState> = HashMap::new();
         let mut db_contracts: HashMap<ContractId, ContractState> = HashMap::new();
+        let mut db_blobs: HashMap<BlobId, BlobState> = HashMap::new();
         let mut db_messages: HashMap<Nonce, MessageState> = HashMap::new();
+
+        if let PoolTransaction::Blob(checked_tx, _) = tx.as_ref() {
+            let blob_id = checked_tx.transaction().blob_id();
+            if db
+                .blob_exist(blob_id)
+                .map_err(|e| Error::Database(format!("{:?}", e)))?
+            {
+                return Err(Error::NotInsertedBlobIdAlreadyTaken(*blob_id))
+            }
+
+            if let Some(state) = self.blobs.get(blob_id) {
+                if state.tip >= tx.tip() {
```

Comment:
This doesn't seem like a "collision" error, but we are returning `NotInsertedCollisionBlobId`

Reply:
If new transaction that deploys the same Blob has lower tip, it is a collision. We can't have two transactions that deploy the same Blob

Reply:
Sure. It's a collision if it's `<` as well though. What we're really doing here is choosing which tx we like more. To the end user the collision error is helpful, I guess, and not inaccurate. `NotInsertedCollisionBlobIdTipLowerThanOtherTx` would be more encompassing. 

I guess I'm having a hard time parsing this function. It would help if it was a little more declarative.

Reply:
It would be nice to change all other places as well with a better name. But it is not part of this PR and will be addressed during the refactoring of the TxPool=) You can check https://github.com/FuelLabs/fuel-core/tree/feature/txpool-v2 it already uses better names.